src/System.Runtime/tests/System/PseudoCustomAttributeTests.cs

```diff
@@ -0,0 +1,152 @@
+                    .Where((e) => e.GetType() == attribute.GetType()).Single());
+                break;
+            case MemberInfo memberInfo:
+                Assert.Equal(attribute, Attribute.GetCustomAttributes(memberInfo, typeof(Attribute))
+                    .Where((e) => e.GetType() == attribute.GetType()).Single());
+                Assert.Equal(attribute, Attribute.GetCustomAttributes(memberInfo, typeof(Attribute), true)
+                    .Where((e) => e.GetType() == attribute.GetType()).Single());
+                break;
+            case ParameterInfo parameterInfo:
+                Assert.Equal(attribute, Attribute.GetCustomAttributes(parameterInfo, typeof(Attribute))
+                    .Where((e) => e.GetType() == attribute.GetType()).Single());
+                Assert.Equal(attribute, Attribute.GetCustomAttributes(parameterInfo, typeof(Attribute), true)
+                    .Where((e) => e.GetType() == attribute.GetType()).Single());
+                break;
+            default:
+                Assert.True(false);
+                break;
+            }
+        }
+
+        private static IEnumerable<object[]> TestData()
+        {
+            yield return new object[] { typeof(TestType), new SerializableAttribute() };
+            
+            Type testComInterface = TryGetComInterfaceType();
+            if (testComInterface != null)
+                yield return new object[] { testComInterface, new ComImportAttribute() };
+
+            FieldInfo testField = typeof(TestType).GetField("_testField");
+
+            yield return new object[] { testField, new FieldOffsetAttribute(120) };
+            yield return new object[] { testField, new NonSerializedAttribute() };
+            yield return new object[] { testField, new MarshalAsAttribute(UnmanagedType.ByValTStr) { SizeConst = 100 } };
+
+            MethodInfo testMethod = typeof(TestType).GetMethod("TestMethod");
+
+            ParameterInfo testMethodParameter = testMethod.GetParameters()[0];
+            yield return new object[] { testMethodParameter, new MarshalAsAttribute(UnmanagedType.LPArray) };
+            yield return new object[] { testMethodParameter, new InAttribute() };
+            yield return new object[] { testMethodParameter, new OutAttribute() };
+            yield return new object[] { testMethodParameter, new OptionalAttribute() };
+
+            yield return new object[] { testMethod.ReturnParameter, new MarshalAsAttribute(UnmanagedType.Bool) };
+
+            yield return new object[] { testMethod, 
+                new DllImportAttribute("nonexistant") { CharSet = CharSet.Unicode, SetLastError = true, EntryPoint = "MyEntryPoint" } };
+
+            yield return new object[] { testMethod, new PreserveSigAttribute() };
+        }
+
+        private static Type TryGetComInterfaceType()
+        {
+            // Ignore the exceptions if COM is not supported
```

Comment:
Interesting. typeof can fail?  I would have thought a failure here would be at JIT time, in which case the try/catch wouldn't matter, but I guess no?

Reply:
I have not actually tested this on Unix and it did not work as you have pointed out.